server/src/main/java/com/hedera/block/server/verification/hasher/CommonUtils.java

```diff
@@ -0,0 +1,181 @@
+ */
+public final class CommonUtils {
+    private CommonUtils() {
+        throw new UnsupportedOperationException("Utility Class");
+    }
+
+    /**
+     * The size of an SHA-384 hash in bytes.
+     */
+    public static final int HASH_SIZE = DigestType.SHA_384.digestLength();
+
+    /**
+     * The tag for the SHA-384 algorithm.
+     */
+    private static final String sha384HashTag = "SHA-384";
+
+    /**
+     * Returns the SHA-384 hash of the given bytes.
+     * @param bytes the bytes to hash
+     * @return the SHA-384 hash of the given bytes
+     */
+    public static Bytes noThrowSha384HashOf(final Bytes bytes) {
+        return Bytes.wrap(noThrowSha384HashOf(bytes.toByteArray()));
+    }
+
+    /**
+     * Returns the SHA-384 hash of the given byte array.
+     * @param byteArray the byte array to hash
+     * @return the SHA-384 hash of the given byte array
+     */
+    public static byte[] noThrowSha384HashOf(final byte[] byteArray) {
+        try {
+            return MessageDigest.getInstance(sha384HashTag).digest(byteArray);
+        } catch (final NoSuchAlgorithmException fatal) {
+            throw new IllegalStateException(fatal);
+        }
+    }
+
+    /**
+     * Returns a {@link MessageDigest} instance for the SHA-384 algorithm, throwing an unchecked exception if the
+     * algorithm is not found.
+     * @return a {@link MessageDigest} instance for the SHA-384 algorithm
+     */
+    public static MessageDigest sha384DigestOrThrow() {
+        try {
+            return MessageDigest.getInstance(DigestType.SHA_384.algorithmName());
+        } catch (final NoSuchAlgorithmException fatal) {
+            throw new IllegalStateException(fatal);
+        }
+    }
+
+    /**
+     * Hashes the given left and right hashes.
+     * @param leftHash the left hash
+     * @param rightHash the right hash
+     * @return the combined hash
+     */
+    public static Bytes combine(@NonNull final Bytes leftHash, @NonNull final Bytes rightHash) {
+        return Bytes.wrap(combine(leftHash.toByteArray(), rightHash.toByteArray()));
+    }
+
+    /**
+     * Hashes the given left and right hashes.
+     * @param leftHash the left hash
+     * @param rightHash the right hash
+     * @return the combined hash
+     */
+    public static byte[] combine(final byte[] leftHash, final byte[] rightHash) {
+        try {
+            final var digest = MessageDigest.getInstance(DigestType.SHA_384.algorithmName());
+            digest.update(leftHash);
+            digest.update(rightHash);
+            return digest.digest();
+        } catch (final NoSuchAlgorithmException fatal) {
+            throw new IllegalStateException(fatal);
+        }
+    }
+
+    /**
+     * Returns the Hashes (input and output) of a list of block items.
+     * @param blockItems the block items
+     * @return the Hashes of the block items
+     */
+    public static Hashes getBlockHashes(List<BlockItemUnparsed> blockItems) {
```

Comment:
could add the non-null annotation to the parameter for the static code analysis

Reply:
✅ Done!